### PR TITLE
Check pressed mouse button in MouseDownHandler

### PR DESCRIPTION
**Pull request by jomarko**
This PR check if the pressed mouse buttons is recognized correctly in the mouse down handler

@romartin would you mind to have a look?

Ensemble:
- #27
- kiegroup/lienzo-core#43

### CODE REVIEW

File: src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java [hunk 1]
```diff
@@ -145,4 +150,49 @@ public void testDragLayer() throws Exception {
 
         verify(mediators, times(5)).handleEvent(any(GwtEvent.class));
     }
+
+    @Test
+    public void testMouseDownHandler_RightButton() {
+        final boolean isLeft = false;
+        final boolean isMidlle = false;
```
Comment (manstis) [inline src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java]: Typo, ```middle``` not ```midlle```

File: src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java [hunk 2]
```diff
@@ -145,4 +150,49 @@ public void testDragLayer() throws Exception {
...
+
+        mouseDownHandler.getValue().onMouseDown(event);
+
+        verify(dragMouseControl).allowDrag(isLeft, isMidlle, isRight);
+    }
+
+    @Test
+    public void testMouseDownHandler_LeftButton() {
+        final boolean isLeft = true;
+        final boolean isMidlle = false;
```
Comment (manstis) [inline src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java]: Typo, ```middle``` not ```midlle```

**Comment by jomarko** [comment]
@manstis thanks, updated.

File: src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java [hunk 3]
```diff
@@ -156,36 +156,26 @@ public void testMouseDownHandler_RightButton() {
...
 
     @Test
     public void testMouseDownHandler_MiddleButton() {
         final boolean isLeft = false;
         final boolean isMiddle = true;
         final boolean isRight = false;
+        verifyMouseDownHandlerButton(isLeft, isMiddle, isRight);
+    }
+
+    private void verifyMouseDownHandlerButton(final boolean isLeft, final boolean isMiddle, final boolean isRight) {
```
Comment (manstis) [inline src/test/java/com/ait/lienzo/client/widget/LienzoHandlerManagerTest.java]: This refactor is only partially complete.

You need to also pass a parameter to setup ```doReturn(NativeEvent.XXX).when(event).getNativeButton()```

**Comment by jomarko** [comment]
@manstis you are too fast :), I introduced also the helper method.

**Pull request merged**